File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change:");
    }
    while (dollars <= 0);
    int cents = round(dollars * 100);
    //printf("%i\n",cents);

    int coinNum = 0;
    while (cents >= 25)
    {
        cents -= 25;
        coinNum += 1;
        //printf("cents:%i\n",cents);
        //printf("coinNum:%i\n",coinNum);
    }
    
    while (cents >= 10)
    {
        cents -= 10;
        coinNum += 1;
        //printf("cents:%i\n",cents);
        //printf("coinNum:%i\n",coinNum);
    }
    
    while (cents >= 5)
    {
        cents -= 5;
        coinNum += 1;
        //printf("cents:%i\n",cents);
        //printf("coinNum:%i\n",coinNum);
    }
    
    while (cents >= 1)
    {
        cents -= 1;
        coinNum += 1;
        //printf("cents:%i\n",cents);
        //printf("coinNum:%i\n",coinNum);
    }
    printf("%i\n", coinNum);
}

File: Shinpei2-cs50-2019-x-sentimental-cash-20190725T152708Z/cash.py
# importing get_float
from cs50 import get_float

# prompt to input cash by float until the value is valid
cash = 0
while True:
    cash = get_float("Change owed: ")
    if 0 < cash:
        break

# cash convert dollars
cash = cash * 100

# count the number of coins
count = 0

# count 25pence coins
while cash >= 25:
    cash -= 25
    count += 1

# count 10pence coins
while cash >= 10:
    cash -= 10
    count += 1

# count 5pence coins
while cash >= 5:
    cash -= 5
    count += 1

# count 1pence coins
while cash >= 1:
    cash -= 1
    count += 1

# print coin counts
print(count)